File: src/main/java/com/vjiki/music/mapper/FollowerMapper.java
package com.vjiki.music.mapper;

import com.vjiki.music.dto.FollowerResponse;
import com.vjiki.music.entity.UserFollow;

public final class FollowerMapper {

    private FollowerMapper() {
    }

    public static FollowerResponse toResponse(UserFollow follow) {
        return FollowerResponse.builder()
                .followerId(follow.getFollowerId())
                .followerEmail(follow.getFollower() == null ? "" : follow.getFollower().getEmail())
                .followerNickname(follow.getFollower() == null ? "" : follow.getFollower().getNickname())
                .followerAvatarUrl(follow.getFollower() == null ? null : follow.getFollower().getAvatarUrl())
                .followedAt(follow.getCreatedAt())
                .build();
    }
}

File: src/main/kotlin/com/vjiki/music/mapper/FollowerMapper.kt
package com.vjiki.music.mapper

import com.vjiki.music.dto.FollowerResponse
import com.vjiki.music.entity.UserFollow
import org.springframework.stereotype.Component

@Component
object FollowerMapper {
    fun UserFollow.toResponse(): FollowerResponse {
        return FollowerResponse(
            followerId = followerId,
            followerEmail = follower?.email ?: "",
            followerNickname = follower?.nickname ?: "",
            followerAvatarUrl = follower?.avatarUrl,
            followedAt = createdAt
        )
    }
}

